hera_pspec/container.py

```diff
@@ -0,0 +1,305 @@
+            pname was not specified).
+        """
+        # Check that group is in keys and extract it if so
+        key1 = "%s" % group
+        if key1 in self.data.keys():
+            grp = self.data[key1]
+        else:
+            raise KeyError("No group named '%s'" % key1)
+        
+        # If psname was specified, check that it exists and extract
+        if psname is not None:
+            key2 = "%s" % psname
+            
+            # Load power spectrum if it exists
+            if key2 in grp.keys():
+                return self._load_pspec(grp[key2])
+            else:
+                raise KeyError("No pspec named '%s' in group '%s'" % (key2, key1))
+        
+        
+        # Otherwise, extract all available power spectra
+        spectra = []
+        def pspec_filter(n, obj):
+            if u'pspec_type' in obj.attrs.keys():
+                spectra.append(self._load_pspec(obj))
+        
+        # Traverse the entire set of groups/datasets looking for pspecs
+        grp.visititems(pspec_filter)
+        return spectra
+        
+    
+    def spectra(self, group):
+        """
+        Return list of available power spectra.
+        
+        Parameters
+        ----------
+        group : str
+            Which group to list power spectra from.
+        
+        Returns
+        -------
+        ps_list : list of str
+            List of names of power spectra in the group.
+        """
+        # Check that group is in keys and extract it if so
+        key1 = "%s" % group
+        if key1 in self.data.keys():
+            grp = self.data[key1]
+        else:
+            raise KeyError("No group named '%s'" % key1)
+        
+        # Filter to look for pspec objects
+        ps_list = []
+        def pspec_filter(n, obj):
+            if u'pspec_type' in obj.attrs.keys():
+                ps_list.append(n)
+        
+        # Traverse the entire set of groups/datasets looking for pspecs
+        grp.visititems(pspec_filter)
+        return ps_list
+    
+    def groups(self):
+        """
+        Return list of groups in the container.
+        
+        Returns
+        -------
+        group_list : list of str
+            List of group names.
+        """
+        groups = self.data.keys()
+        if u'header' in groups: groups.remove(u'header')
+        return groups
+    
+    def tree(self):
+        """
+        Output a string containing a tree diagram of groups and the power 
+        spectra that they contain.
+        """
+        s = ""
+        for grp in self.groups():
+            s += "(%s)\n" % grp
+            for pspec in self.spectra(grp):
+                s += "  |--%s\n" % pspec
+        return s
+    
+    def save(self):
+        """
+        Force HDF5 file to flush to disk.
+        """
+        self.data.flush()
+    
+    def __del__(self):
+        """
+        Make sure that HDF5 file is closed on destruct.
+        """
+        try:
```

Comment:
why is there a `try: except:` here? why would this every fail? If this is needed, can we be specific as to what Error is will except?